src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -360,6 +360,36 @@ try
         _api.dirtyRect = til::rect{ 0, _api.invalidatedRows.x, _api.cellCount.x, _api.invalidatedRows.y };
     }
 
+    // This is an important block of code for our TileHashMap.
+    // We only process glyphs within the dirtyRect, but glyphs outside of the
+    // dirtyRect are still in use and shouldn't be discarded. This is critical
+    // if someone uses a tool like tmux to split the terminal horizontally.
+    // If they then print a lot of Unicode text on just one side, we have to
+    // ensure that the (for example) plain ASCII glyphs on the other half of the
+    // viewport are still retained. This bit of code "refreshes" those glyphs and
+    // brings them to the front of the LRU queue to prevent them from being reused.
+    {
+        std::array<til::point, 2> ranges{ {
+            { 0, _api.dirtyRect.top },
+            { _api.dirtyRect.bottom, _api.cellCount.y },
+        } };
+        const auto stride = static_cast<size_t>(_r.cellCount.x);
+        for (const auto& p : ranges)
+        {
+            const auto from = p.x;
+            const auto to = p.y;

```

Comment:
Sorry, I'm just really confused about this. Why are we iterating from the x-coord to the y-coord? The inner loop makes sense because we're looping over each y value. Shouldn't this be looping over the x values then?

Reply:
I've added a brief comment explaining this peculiar code.